crux/src/Crux/Report.hs

```diff
@@ -84,10 +86,17 @@ maybeGenerateSource opts files =
 -- | Return a list of all program locations referenced in a set of
 -- proved goals.
 provedGoalLocs :: ProvedGoals -> [ProgramLoc]
-provedGoalLocs (Branch goals1 goals2) = provedGoalLocs goals1 ++
-                                        provedGoalLocs goals2
-provedGoalLocs (ProvedGoal _ err locs _) = locs ++ [simErrorLoc err]
-provedGoalLocs (NotProvedGoal _ err _ locs _) = locs ++ [simErrorLoc err]
+provedGoalLocs = concatMap Fold.toList . provedGoalTraces
+
+-- | Return a list of all of the traces referenced in a set of proved goals.
+provedGoalTraces :: ProvedGoals -> Seq (Seq ProgramLoc)
```

Comment:
Why do you use nested `Seq`s here? Perhaps you're trying to preserve some structure here, but if so, the structure should be documented.

Reply:
Documented, let me know if that makes things more clear.

Reply:
Thanks, that's helpful.